util/tls/tls.go

```diff
@@ -139,12 +142,31 @@ func GetServerTLSConfigFromSecret(ctx context.Context, kubectlConfig kubernetes.
 		return nil, err
 	}
 
+	// Join certpem and keypem into an X509KeyPair
 	cert, err := tls.X509KeyPair(certpem, keypem)
 	if err != nil {
 		return nil, err
 	}
 
+	// Retrieve the System Certificate Pool
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil {
+		log.Warnf("failed to get system certificate pool: %v, continuing with empty certificate trust", err)
+		rootCAs = x509.NewCertPool()
+	}
+
+	// Pull the ca.crt from the Kubernetes secret
+	capem, err := util.GetSecrets(ctx, kubectlConfig, namespace, tlsKubernetesSecretName, tlsCaSecretKey)
+	if err == nil {
+		if !rootCAs.AppendCertsFromPEM(capem) {
+			log.Warn("failed to append ca.crt to the trusted CA pool")
+		}
+	} else {
+		log.Warnf("skipped adding ca.crt to local certificate trusts: %v", err)
+	}
+
 	return &tls.Config{
+		RootCAs:      rootCAs,
```

Comment:
The reason I've added this is because I believe the HTTP -> grpc gateway actually tries to validate the GRPC server certificate with the config [here](https://github.com/argoproj/argo-workflows/blob/4664aeac4ffa208114b8483e6300c39b537b402d/server/apiserver/argoserver.go#L311)

As I am using a private CA, the certificate validation fails internally. Adding the provided CA to the trusts means that the grpc gateway can connect (to itself..). 

This doesn't occur when using the in-memory certificate, because the inmem cert sets `InsecureSkipVerify` to [true](https://github.com/argoproj/argo-workflows/blob/master/util/tls/tls.go#L127).

No worries about missing the flag!

Reply:
@ChaosInTheCRD, just want to bump this one